modules/apps/depot/depot-test/src/testFunctional/tests/DepotWCIntegration.testcase

```diff
@@ -112,6 +112,90 @@ definition {
 		WebContent.viewPGViaWCD(webContentContent = "WC WebContent Depot 2");
 	}
 
+	@description = "This ensures that a depot web content can be selected through the fragment in a content page on a connected site."
+	@priority = "5"
+	test SelectFromFragmentInContentPage {
+		DepotNavigator.openDepotAdmin();
+
+		DepotNavigator.gotoDepot(
+			depotName = "Test Depot Name",
+			portlet = "Web Content");
+
+		WebContentNavigator.gotoAddCP();
+
+		WebContent.addCP(
+			webContentContent = "WC WebContent Depot",
+			webContentTitle = "WC WebContent Title");
+
+		PortletEntry.publish();
+
+		DepotNavigator.openDepotAdmin();
+
+		DepotNavigator.gotoEdit(depotName = "Test Depot Name");
+
+		DepotNavigator.gotoSites(depotName = "Test Depot Name");
+
+		Depot.connectSite(siteList = "Site Name");
+
+		SitePages.openPagesAdmin(siteURLKey = "site-name");
+
+		ContentPages.addPage(pageName = "Test Content Page Name");
+
+		PageEditor.addContentDisplay(fragmentName = "Content");
+
+		PageEditor.editContentDisplay(
+			depotName = "Test Depot Name",
+			webcontent = "true",
+			webContentContent = "WC WebContent Depot",
+			webContentTitle = "WC WebContent Title");
+
+		PageEditor.clickPublish();
+
+		Navigator.openWithAppendToBaseURL(urlAppend = "web/site-name/test-content-page-name");
+
+		AssertTextEquals.assertPartialTextAceEditor(
+			locator1 = "PageEditor#CONTENT_PAGE_TEXT_CONTENT",
+			value1 = "WC WebContent Depot");
+	}
+
+	@description = "This ensures that a depot web content can be selected through the fragment in a display page template on a connected site."
+	@priority = "5"
+	test SelectFromFragmentInDisplayPageTemplate {
+		DepotNavigator.openDepotAdmin();
+
+		DepotNavigator.gotoDepot(
+			depotName = "Test Depot Name",
+			portlet = "Web Content");
+
+		WebContentNavigator.gotoAddCP();
+
+		WebContent.addCP(
+			webContentContent = "WC WebContent Depot",
+			webContentTitle = "WC WebContent Title");
+
+		PortletEntry.publish();
+
+		DepotNavigator.openDepotAdmin();
+
+		DepotNavigator.gotoEdit(depotName = "Test Depot Name");
+
+		DepotNavigator.gotoSites(depotName = "Test Depot Name");
+
+		Depot.connectSite(siteList = "Site Name");
+
+		DisplayPageTemplatesAdmin.openDisplayPagesAdmin(siteURLKey = "site-name");
+
+		DisplayPageTemplatesAdmin.addDisplayPage(displayPageName = "Display Page Name");
+
+		PageEditor.addContentDisplay(fragmentName = "Content");
+
+		PageEditor.editContentDisplay(
+			depotName = "Test Depot Name",
+			webcontent = "true",
+			webContentContent = "WC WebContent Depot",
+			webContentTitle = "WC WebContent Title");
+	}
```

Comment:
@liuxingye we should also publish this and view the page template with the web content. Can you create a test improvement for it?

Reply:
Sure, I'll do it here ([LRQA-54647](https://issues.liferay.com/browse/LRQA-54647)) once the new test is merged to upstream.